dll/win32/shell32/utils.h

```diff
@@ -124,3 +125,28 @@ struct ClipboardViewerChain
         return 0;
     }
 };
+
+struct CCidaChildArrayHelper
+{
+    // Note: This just creates an array pointing to the items and has the same lifetime as the CIDA.
+    // Use _ILCopyCidaToaPidl if you need the items to outlive the CIDA!
+    explicit CCidaChildArrayHelper(const CIDA *pCida)
+    {
+        m_hr = E_OUTOFMEMORY;
+        m_array = (PCUIDLIST_RELATIVE_ARRAY)SHAlloc(pCida->cidl * sizeof(LPITEMIDLIST));
+        if (m_array)
+        {
+            m_hr = S_OK;
+            for (UINT i = 0; i < pCida->cidl; ++i)
+                *(LPITEMIDLIST*)(&m_array[i]) = (LPITEMIDLIST)HIDA_GetPIDLItem(pCida, i);
+        }
+    }
+    ~CCidaChildArrayHelper() { SHFree((LPITEMIDLIST*)m_array); }
+
+    HRESULT hr() const { return m_hr; }
+    PCUIDLIST_RELATIVE_ARRAY GetItems() const { return m_array; }
+
+    HRESULT m_hr;
+    PCUIDLIST_RELATIVE_ARRAY m_array;
+};
+#endif
```

Comment:
```suggestion
#endif // __cplusplus
```
so that we know when we're down there what this endif was referring to.